### PR TITLE
Sindhu/gathernd

### PR DESCRIPTION
**Pull request by sindhu-nervana**

### CODE REVIEW

File: test/test_array_ops.cpp [hunk 1]
```diff
@@ -316,6 +316,55 @@ TEST(ArrayOps, ExpandDims) {
...
+
+  OpExecuter opexecuter(root, "GatherNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+
+}  // end of test op GatherND
+
+// Test op:GatherND
+TEST(ArrayOps, GatherNd_3D) {
```
Comment (sayantan-nervana) [inline test/test_array_ops.cpp]: Please remove underscore from gtest names. apparently gtest reserves the right to use underscores willy-nilly

**Comment by sindhu-nervana** [comment]
Two gather_nd tf python tests are failing, so disabled those. Debugging in parallel.

File: test/test_array_ops.cpp [hunk 2]
```diff
@@ -316,6 +316,55 @@ TEST(ArrayOps, ExpandDims) {
...
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "GatherNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+
+}  // end of test op GatherND
+
 // Test op: Gather. vector indices
```
Comment (sindhu-nervana) [inline test/test_array_ops.cpp]: These C++ tests will fail on GPU as GatherNd is not supported on GPU. `buildkite/ngtf-r0-dot-13-cpu-ubuntu-18-04`

File: src/ngraph_mark_for_clustering.cc [hunk 3]
```diff
@@ -439,6 +440,9 @@ Status MarkForClustering(Graph* graph,
       // DT_FLOAT
       type_constraint_map["FusedBatchNormV2"]["T"] = {DT_FLOAT};
       type_constraint_map["FusedBatchNormGrad"]["T"] = NGraphNumericDTypes();
+      type_constraint_map["GatherNd"]["Tparams"] = {
+          DT_FLOAT};  // NGraphDTypes();
```
Comment (sayantan-nervana) [inline src/ngraph_mark_for_clustering.cc]: Can we expand this to more types?

File: test/test_array_ops.cpp (hunk 2 above)
Comment (sayantan-nervana) [reply test/test_array_ops.cpp]: To pass CI on GPU, please add this test to the filter here: https://github.com/tensorflow/ngraph-bridge/blob/master/test/ci/buildkite/test_runner.py#L97

File: src/ngraph_mark_for_clustering.cc (hunk 3 above)
Comment (sindhu-nervana) [reply src/ngraph_mark_for_clustering.cc]: TF python tests were throwing unsupported CPU builder errors for more types.

Comment (sindhu-nervana) [reply src/ngraph_mark_for_clustering.cc]: TODO:   {DT_FLOAT} passes, but the full list `NGraphDTypes()` fails, so may need to check what other datatypes will pass

**Comment by sindhu-nervana** [comment]
@avijit-nervana Looks like the failure was introduced after TF 1.14 change and further. I am bisecting now to figure out which commit might've introduced it to fix.

**Pull request merged**